### PR TITLE
fix test swap.c for watcom …

### PR DESCRIPTION
**Pull request by jmalak**

### CODE REVIEW

File: src/tests/swap.c [hunk 1]
```diff
@@ -278,7 +275,7 @@ void test_swap_speed2 (const char *buf)
   TIME_IT (intrin_byteswap,      (buf, swap_size/4), loops);
 }
 
-void __watcall sigill_handler2 (int sig)
+void sigill_handler2 (int sig)
```
Comment (gvanem) [inline src/tests/swap.c:278]: Strange. According to `signal.h`, a Watcom signal-handler is:
`typedef void (_WCCALLBACK *__sig_func)( int );` and  `_WCCALLBACK` is `__watcall`.

But dropping `__watcall`, it works.

File: src/tests/makefile.all [hunk 2]
```diff
@@ -340,33 +346,38 @@ NO_W32_CFLAGS = -D_Windows
   endef
 
 @elifdef WATCOM
-  LDFLAGS = option quiet, map, eliminate, caseexact
+  LDFLAGS = option quiet, map, eliminate
```
Comment (gvanem) [inline src/tests/makefile.all:349]: Why dropping `caseexact`? It's important not to mix symbols with different casing IMHO.

File: src/tests/configur.bat [hunk 3]
```diff
@@ -123,16 +123,20 @@ goto next
...
-echo   "make -f watcom_l.mak" for large model
-echo   "make -f watcom_f.mak" for flat model
-echo   "make -f watcom_x.mak" for flat/X32VM model
+echo   "make -f watcom_s.mak" for small model (16-bit)
+echo   "make -f watcom_l.mak" for large model (16-bit)
+echo   "make -f watcom_3.mak" for small model (32-bit DOS4GW)
+echo   "make -f watcom_f.mak" for flat model  (DOS4GW)
+echo   "make -f watcom_x.mak" for flat model  (X32VM)
 echo   "make -f watcom_w.mak" for Win32
+echo   "make -f watcom_d.mak" for Win32 DLL
```
Comment (gvanem) [inline src/tests/configur.bat:139]: Using a .dll in tests will hardly work. Have you tried it?

**Comment by gvanem** [comment]
Merged. Thanks!

File: src/tests/makefile.all (hunk 2 above)
Comment (jmalak) [reply src/tests/makefile.all:349]: I will submit other fixes to watcom make files to fix some compiler and linker illogical options etc.
If you don't specify your own target (not supported by watcom) then you should use standard `system` directive as first. Important options and directives are handled by `system` directive.
After this directive you can modify anything if you need to change default setup by system directive. Now only 16-bit windows target use case insensitive symbols comparision/selection all others are case sensitive.

File: src/tests/configur.bat (hunk 3 above)
Comment (jmalak) [reply src/tests/configur.bat:139]: I think for WIN32 it should be done as current scenario. I plan to do it, it is preparation for it.

File: src/tests/swap.c (hunk 1 above)
Comment (jmalak) [reply src/tests/swap.c:278]: _WCCALLBACK is not only about calling convention.
__watcall is default that not need to be specified, it has sense if you change default calling convention what is not current and not your case

Comment (gvanem) [reply src/tests/swap.c:278]: So if I do:
```diff
diff --git a/src/tests/swap.c b/src/tests/swap.c
index 07fc157..fad93d5 100644
--- a/src/tests/swap.c
+++ b/src/tests/swap.c
@@ -22,7 +22,7 @@
   #define TEST_UNDEF_OPCODE

 #elif defined(__WATCOMC__) && defined(__386__)
-//#define INVD_CACHE
+  #define INVD_CACHE
   #define TEST_UNDEF_OPCODE
 #endif
```
Compile and run `swap.exe`, I get an exception. Either the `sigill_handler()` in `sysdep.h` is wrong or this:
```c
  #pragma aux _invd_cache = \
          "db 0Fh, 08h";
```

is wrong. But Im not familar with Watcom ASM.

**Pull request merged**